src/components/hooks/use-lazy-video.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from 'react'
+
+type VideoType = {
+    isIntersecting: boolean
+    target: HTMLVideoElement
+}
+
+type VideoSourceType = {
+    tagName: string
+    src?: string
+    dataset?: HTMLVideoElement
+}
+
+const createObserver = (lazyVideos) => {
```

Comment:
Hello @nikitk,

Is it possible to give this parameter a type array, since this is typescript?

Reply:
Hello @jim-deriv 👋 
It's possible and pretty cool idea. Thanks 🙇‍♂️